app/models/products/user_note_mode.rb

```diff
@@ -0,0 +1,49 @@
+module Products
+
+  class UserNoteMode
```

Comment:
This is an interesting pattern.  It still feels a little weird to me--I gotta watch that tapas ep--but the goal of keeping bloat out of Product is a worthy one.

`.valid_values` still feels a little odd though. If your intent is for it to be essentially private, I might expect that to be a constant. Can you add a comment about your intent for `.values` to be the public-facing method?